subscription-engine/kafka-ws/app/Redis/RedisExpireMgmt.js

```diff
@@ -178,10 +181,10 @@ class RedisExpireMgmt {
   /**
    * End connection with redis
    */
-  end() {
+  async end() {
     this.clients.sub.unsubscribe();
-    this.clients.pub.quit();
-    this.clients.sub.quit();
+    await this.clients.pub.quit();
+    await this.clients.sub.quit();
```

Comment:
Attention to detail, `await` is useless here, since `quit` does not return a `Promise`, this should be done via callback parameter to `quit` (as in `RedisManager.shutdownProcess()`).
The lack of a `resolved` Promise _after the execution of a callback_ can result in the problem of the _event loop_ still having tasks to perform at the end of the graceful shutdown... 
In my opinion, this should be improved.